src/_guides/language/concurrency/index.md

```diff
@@ -312,107 +311,144 @@ to implement isolates.
+  // Read some data.
+  final jsonData = await Isolate.run(() async {
+    final fileData = await File(filename).readAsString();
+    final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
+    return jsonData;
+  });
+
+  // Use that data.
+  print('Number of JSON keys: ${jsonData.length}');
 }
 ```
 
-The relevant statement is the last one, which exits the isolate,
-sending `jsonData` to the passed-in `SendPort`.
-Message passing using `SendPort.send` normally involves data copying,
-and thus can be slow.
-However, when you send the data using `Isolate.exit()`,
-then the memory that holds the message in the exiting isolate isn’t copied,
-but instead is transferred to the receiving isolate.
-The sender will nonetheless perform a verification pass to ensure
-the objects are allowed to be transferred.
-
-
-{{site.alert.version-note}}
-  `Isolate.exit()` was added in 2.15.
-  Previous releases support only explicit message passing,
-  using `Isolate.send()` as shown in the next section's example.
-{{site.alert.end}}
-
-The following figure illustrates the communication between
-the main isolate and the worker isolate:
+This example accomplishes the same as the previous:
+a new isolate spawns and executes a computation, sending back the result.
 
-![A figure showing the previous snippets of code running in the main isolate and in the worker isolate](/guides/language/concurrency/images/isolate-api.png)
+However, now the isolate sends a [closure][]. 
+This allows versatility not only for how the isolate can be utilized
+(as closures are less limited than typical named functions),
+but also for the written style of the code.
+Here, `Isolate.run()` executes what looks like local code, concurrently.
+In that sense, you can imagine `run()` almost like a [control flow operator][]
+for “run in parallel”.
 
+[closure]: /guides/language/language-tour#anonymous-functions
+[control flow operator]: /guides/language/language-tour#control-flow-statements
 
 ### Sending multiple messages between isolates
 
-If you need more communication between isolates,
-then you need to use the [`send()` method][] of `SendPort`.
+`Isolate.run()` abstracts a handful of lower-level, 
+isolate-related API to simplify isolate management:
+
+* [`Isolate.spawn()`][] and [`Isolate.exit()`][]
+* [`ReceivePort`][] and [`SendPort`][]
+
+[`Isolate.exit()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/exit.html
+[`Isolate.spawn()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawn.html
+[`ReceivePort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/ReceivePort-class.html
+[`SendPort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/SendPort-class.html
+
+However, there are cases where it’s necessary
+to directly utilize the primitives that make up `run()`
+for more granular control over isolate management.
+Sending multiple messages between isolates is one of those cases.
```

Comment:
**suggestion:** Don't bury the lede. Also, simplify the language. (That is, "control" and "management" are synonyms.)
```suggestion
To provide more granular control of isolates might
require using the primitives that make up `run()`.
One example use case would be sending multiple
messages between isolates.
```